Catch when nodejs request is aborted

M4RC3L05 opened the pull request:
When we receive a request, we start listening to the close event. When fired, we check if the request was aborted using `incoming.aborted`, if it was, we dispatch an abort event to the existing request abort signal.

Without this, when the client aborts the request, the signal on the request was not being called with the abort event.

Closes: https://github.com/honojs/node-server/issues/107

M4RC3L05 commented on src/listener.ts:
```diff
@@ -143,6 +143,13 @@ export const getRequestListener = (
 
+    // Detect if request was aborted.
+    incoming.socket.on('close', () => {
+      if (incoming.aborted) {
```
From the nodejs docs it appears that the `aborted` property is deprecated, and as a replacement, we should check the `destroyed` (https://nodejs.org/dist/latest-v20.x/docs/api/http.html#messageaborted).

For some reason if i use it, it makes most of the tests fail, do not know if this is something related with supertest...

M4RC3L05 replied:
Resolved, had to listen to the close event on the response and check the `incoming.socket.destroyed`

M4RC3L05 commented on test/listener.test.ts:
```diff
@@ -89,3 +89,42 @@ describe('Error handling - async fetchCallback', () => {
   })
 })
+
+describe('Abort request', () => {
```
Is there any more tests that i should include?

yusukebe commented:
Hi @M4RC3L05 

Thank you for the PR. Looks good to me.

@usualoma Could you too review this?

usualoma commented:
I'm preparing an improvement plan now, so wait a minute.

usualoma commented:
Hi @M4RC3L05 
Thank you. The implementation policy makes sense to me!

I hope to achieve the following two points as well.

* The `aborted` property would be `true`.
　 * https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/aborted
* not use `setInterval()` in tests.

As an example, I would like to have something like the following PR, can you incorporate it into your PR?
https://github.com/honojs/node-server/pull/142

M4RC3L05 commented:
Hi, sure, I can work on making does changes

M4RC3L05 commented:
Updated with the improvements pointed out on https://github.com/honojs/node-server/pull/142 (thx btw :pray:).

Also, added some more tests, please @usualoma @yusukebe let me know if they are relevant or not.

M4RC3L05 commented on test/request.test.ts:
```diff
@@ -40,6 +40,34 @@ describe('Request', () => {
       expect(req.url).toBe('http://localhost/foo.txt')
     })
+
+    it('should generate only one `AbortController` per `Request` object created', async () => {
```
Just a sanity check to make sure we do not create a new AbortController every time we access it, or any other property since it calls `getRequestCache`.

usualoma commented:
@M4RC3L05 Thank you. Looks good!

yusukebe commented:
@M4RC3L05 @usualoma Thanks both!

I'd like to merge, as @usualoma as co-author.

@M4RC3L05 
One thing, could run `yarn lint:fix && yarn format:fix` to fix the CI error?

M4RC3L05 commented:
Done

yusukebe commented:
@M4RC3L05 Thanks!!
